metric_learn/sdml.py

```diff
@@ -64,17 +64,25 @@ def _fit(self, pairs, y):
     # set up prior M
     if self.use_cov:
       X = np.vstack({tuple(row) for row in pairs.reshape(-1, pairs.shape[2])})
-      prior = pinvh(np.cov(X, rowvar = False))
+      prior = pinvh(np.atleast_2d(np.cov(X, rowvar=False)))
     else:
       prior = np.identity(pairs.shape[2])
     diff = pairs[:, 0] - pairs[:, 1]
     loss_matrix = (diff.T * y).dot(diff)
     emp_cov = pinvh(prior) + self.balance_param * loss_matrix
-    M, _, _, _, _, _ = quic(emp_cov, lam=self.sparsity_param, msg=self.verbose,
-                            Theta0=np.eye(pairs.shape[2]),
-                            Sigma0=np.eye(pairs.shape[2]))
 
-    self.transformer_ = transformer_from_metric(M)
+    # our initialization will be the matrix with emp_cov's eigenvalues,
```

Comment:
This is an init that we talked about with @bellet, that I found worked better (allowed tests to pass when with identity I had a lot of Linalg Error)